app/services/webpage_fetch_service.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class WebpageFetchService
+  def initialize(url, user_agent)
+    @uri = URI(url)
+    @user_agent = user_agent
+  end
+
+  def call
+    result = HTTParty.get(@uri, { headers: { 'User-Agent' => @user_agent } })
+
+    return false unless valid_result? result
+
+    result
+  rescue HTTParty::Error, Timeout::Error, SocketError
+    false
```

Comment:
```suggestion
    return unless valid_result? result

    result
  rescue HTTParty::Error, Timeout::Error, SocketError
    nil
```

Let's not return `result || nil`: Either we have a boolean method that returns `true || false`, either we have an object method that results `object || nil` ;-)

Reply:
(tests will need adjustment accordingly ;-) )